From ec7d9197d2b0b3f8ab324b235e4783e82d7af7d6 Mon Sep 17 00:00:00 2001
From: Valentin Raevsky <valentin@compulab.co.il>
Date: Wed, 26 Nov 2025 14:47:05 +0200
Subject: [PATCH] compulab: iot-gate-imx8plus: fec: update drive strength for
 TX signals

Update the drive strength value from X6 to X2 for all TX signals of the FEC
interface to optimize signal integrity and meet design constraints

Signed-off-by: Ilya Ledvich <ilya@compulab.co.il>
---
 .../dts/compulab/ucm-imx8m-plus-pinctrl.dtsi  | 28 +++++++++----------
 1 file changed, 14 insertions(+), 14 deletions(-)

diff --git a/arch/arm64/boot/dts/compulab/ucm-imx8m-plus-pinctrl.dtsi b/arch/arm64/boot/dts/compulab/ucm-imx8m-plus-pinctrl.dtsi
index f782690243f4..cc63d5ee755d 100644
--- a/arch/arm64/boot/dts/compulab/ucm-imx8m-plus-pinctrl.dtsi
+++ b/arch/arm64/boot/dts/compulab/ucm-imx8m-plus-pinctrl.dtsi
@@ -6,20 +6,20 @@
 &iomuxc {
 	pinctrl_fec: fecgrp {
 		fsl,pins = <
-			MX8MP_IOMUXC_SD1_CLK__ENET1_MDC					0x3
-			MX8MP_IOMUXC_SD1_CMD__ENET1_MDIO				0x3
-			MX8MP_IOMUXC_SD1_DATA2__ENET1_RGMII_RD0				0x91
-			MX8MP_IOMUXC_SD1_DATA3__ENET1_RGMII_RD1				0x91
-			MX8MP_IOMUXC_SAI1_RXD6__ENET1_RGMII_RD2				0x91
-			MX8MP_IOMUXC_SAI1_RXD7__ENET1_RGMII_RD3				0x91
-			MX8MP_IOMUXC_SAI1_TXC__ENET1_RGMII_RXC				0x91
-			MX8MP_IOMUXC_SAI1_TXFS__ENET1_RGMII_RX_CTL			0x91
-			MX8MP_IOMUXC_SD1_DATA1__ENET1_RGMII_TD0				0x1f
-			MX8MP_IOMUXC_SD1_DATA0__ENET1_RGMII_TD1				0x1f
-			MX8MP_IOMUXC_SAI1_TXD2__ENET1_RGMII_TD2				0x1f
-			MX8MP_IOMUXC_SAI1_TXD3__ENET1_RGMII_TD3				0x1f
-			MX8MP_IOMUXC_SAI1_TXD4__ENET1_RGMII_TX_CTL			0x1f
-			MX8MP_IOMUXC_SAI1_TXD5__ENET1_RGMII_TXC				0x1f
+			MX8MP_IOMUXC_SD1_CLK__ENET1_MDC					0x2
+			MX8MP_IOMUXC_SD1_CMD__ENET1_MDIO				0x2
+			MX8MP_IOMUXC_SD1_DATA2__ENET1_RGMII_RD0				0x90
+			MX8MP_IOMUXC_SD1_DATA3__ENET1_RGMII_RD1				0x90
+			MX8MP_IOMUXC_SAI1_RXD6__ENET1_RGMII_RD2				0x90
+			MX8MP_IOMUXC_SAI1_RXD7__ENET1_RGMII_RD3				0x90
+			MX8MP_IOMUXC_SAI1_TXC__ENET1_RGMII_RXC				0x90
+			MX8MP_IOMUXC_SAI1_TXFS__ENET1_RGMII_RX_CTL			0x90
+			MX8MP_IOMUXC_SD1_DATA1__ENET1_RGMII_TD0				0x14
+			MX8MP_IOMUXC_SD1_DATA0__ENET1_RGMII_TD1				0x14
+			MX8MP_IOMUXC_SAI1_TXD2__ENET1_RGMII_TD2				0x14
+			MX8MP_IOMUXC_SAI1_TXD3__ENET1_RGMII_TD3				0x14
+			MX8MP_IOMUXC_SAI1_TXD4__ENET1_RGMII_TX_CTL			0x14
+			MX8MP_IOMUXC_SAI1_TXD5__ENET1_RGMII_TXC				0x16
 		>;
 	};
 
-- 
2.34.1

